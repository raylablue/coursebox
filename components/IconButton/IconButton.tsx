import { FC, MouseEvent } from "react";
import styled from "@emotion/styled";
import { css } from "@emotion/react";
import { boxShadow, transition } from "../styles";
import { Icon, Props as IconProps } from "../Icon";

type ButtonProps = { size: string };

const Button = styled.button`
  all: unset;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  ${({ size }) => css`
    width: ${size};
    height: ${size};
  `}
  border-radius: 50%;

  ${transition()};

  ${({ theme }) =>
    boxShadow(theme.components.shadow1, theme.components.shadow2)}

  &:active {
    ${({ theme }) =>
      boxShadow(theme.components.shadow1, theme.components.shadow2, true)}
  }

  &:hover {
    opacity: 0.9;
  }
`;

export type Props = {
  /** onClick callback */
  onClick: (event: MouseEvent<HTMLButtonElement>) => void;
} & IconProps;

export const IconButton: FC<Props> = ({ onClick, ...props }) => (
  <Button
    onClick={onClick}
    size={`${(props.size || 2) * 2}rem`}
    title={props.name}
  >
    <Icon {...props} />
  </Button>
);
