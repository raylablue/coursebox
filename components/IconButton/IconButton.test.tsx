import userEvent from "@testing-library/user-event";

import { render, screen } from "@/test-utils";

import { IconButton } from "./IconButton";

describe("IconButton test cases", () => {
  it("Render check", () => {
    const onClick = jest.fn();
    const { asFragment } = render(
      <IconButton name="Home" onClick={onClick}>
        Button
      </IconButton>
    );

    expect(asFragment()).toMatchSnapshot();
  });
  it("Check onClick callback", async () => {
    const onClick = jest.fn();

    render(
      <IconButton name="Home" onClick={onClick}>
        Button
      </IconButton>
    );

    const element = screen.getByRole("button");

    await userEvent.click(element);

    expect(onClick).toHaveBeenCalled();
  });
});
