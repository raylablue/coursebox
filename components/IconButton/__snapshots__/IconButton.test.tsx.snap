// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton test cases Render check 1`] = `
<DocumentFragment>
  <button
    class="css-1176sk3"
    title="Home"
  >
    <div
      class="css-14g5jgk"
      size="2"
    >
      <svg
        fill="currentColor"
        height="2rem"
        viewBox="0 0 1024 1024"
        width="2rem"
      >
        <path
          d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
        />
      </svg>
    </div>
  </button>
</DocumentFragment>
`;
